File: 2015/day_1/not_quite_lisp_part_two.py
def find_floor(instructions):
	current_floor = []
	patterns = { "(": ")", ")": "(" }
	for pos, instruction in enumerate(instructions):
		if current_floor and current_floor[-1] == patterns[instruction]:
			current_floor.pop()
		else:
			if instruction == ")":
				return pos + 1 
			current_floor.append(instruction)
			
	return -1

with open("input") as f:
	lines = f.readlines()
	print(find_floor(lines[0].strip()))

File: 2015/day_1/not_quite_lisp_part_two.rs
use std::{
  collections::HashMap,
  fs::File,
  io::Read
};

fn find_floor(instructions: String) -> i64 {
  let mut current_floor = vec![];
  let patterns = HashMap::from([
    ('(', ')'),
    (')', '('),
  ]);
  for (pos, instruction) in instructions.chars().enumerate() {
    if current_floor.last() == Some(&patterns[&instruction]) {
      current_floor.pop();
    } else {
      if instruction == ')' {
        return (pos + 1) as i64;
      }
      current_floor.push(instruction);
    }
  }
  -1
}

fn main() {
  let mut file = File::open("./input").unwrap();
  let mut buf = String::from("");
  file.read_to_string(&mut buf).unwrap();
  println!("{}", find_floor(String::from(buf.trim_end())));
}